dev_tools/clang_format.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+# Copyright 2023 The Magma Authors.
+
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script formats all C/C++ files in the magma repo using clang-format-11.
+# The script can be executed on the dev VM or on the
+# host machine if clang-format-11 is installed.
```

Comment:
Careful, if on the host VM `MAGMA_ROOT` won't be defined. It might be best to not support this for now.

Reply:
Or you can ask for the user to set the variable correctly.

Reply:
Regarding "on the host VM", do you mean the host?

Reply:
Done